src/layer.rs

```diff
@@ -123,6 +126,7 @@ pub struct LayerData {
     pub(crate) blend_mode: BlendMode,
     pub(crate) opacity: u8,
     pub(crate) layer_type: LayerType,
+    pub(crate) tilemap_index: Option<u32>,
```

Comment:
It might be better to put this data under the `LayerType::Tilemap` enum. In that case the Layer struct would need a little more refactoring.